crates/task-impls/src/quorum_vote/mod.rs

```diff
@@ -110,31 +111,41 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES> + 'static, V: Versions>
             .cloned();
         maybe_parent = match maybe_parent {
             Some(p) => Some(p),
-            None => fetch_proposal(
-                justify_qc.view_number(),
-                self.sender.clone(),
-                self.receiver.activate_cloned(),
-                Arc::clone(&self.quorum_membership),
-                OuterConsensus::new(Arc::clone(&self.consensus.inner_consensus)),
-                self.public_key.clone(),
-                self.private_key.clone(),
-                &self.upgrade_lock,
-            )
-            .await
-            .ok(),
+            None => {
```

Comment:
Can we consolidate the consensus map lookups to this function, So it's easier to see where we are getting all the information.  If we're already looking up the parent here, maybe we don't need to do the `validated_view` thing in the calling function at all